### PR TITLE
Added code to read depth scale from camera

### PR DESCRIPTION
**Pull request by reaganlo**
**Please ensure the above _guidelines for contributing_ are met.**

Fixes Issue: #46 

Changes proposed in this pull request:
- Implement #46 after re-basing and refactoring based on latest code.

### CODE REVIEW

File: realsense_camera/src/base_nodelet.cpp [hunk 1]
```diff
@@ -554,10 +556,16 @@ namespace realsense_camera
     if (stream_index == RS_STREAM_DEPTH)
     {
       // fill depth buffer
-      image_depth16_ = reinterpret_cast <const uint16_t * >(rs_get_frame_data(rs_device_, RS_STREAM_DEPTH, 0));
+      image_depth16_ = reinterpret_cast <const uint16_t * >(rs_get_frame_data(rs_device_, stream_index, 0));
```
Comment (mdhorn) [inline realsense_camera/src/base_nodelet.cpp]: Can we add a check to see if depth_scale_meters_ == MILLIMETER_METERS, and not call the covert wrapper and just make the assignment instead?

Comment (mdhorn) [inline realsense_camera/src/base_nodelet.cpp]: Please remove the whitespace from the cast.
`reinterpret_cast<const uint16_t *>`

**Comment by reaganlo** [comment]
@mdhorn Updated review comments

**Comment by mdhorn** [comment]
Approved 👍

File: realsense_camera/src/base_nodelet.cpp [hunk 2]
```diff
@@ -554,10 +556,23 @@ namespace realsense_camera
     if (stream_index == RS_STREAM_DEPTH)
     {
       // fill depth buffer
-      image_depth16_ = reinterpret_cast <const uint16_t * >(rs_get_frame_data(rs_device_, RS_STREAM_DEPTH, 0));
+      image_depth16_ = reinterpret_cast <const uint16_t *>(rs_get_frame_data(rs_device_, stream_index, 0));
```
Comment (rjingar) [inline realsense_camera/src/base_nodelet.cpp]: Please remove the whitespace between reinterpret_cast and <const...

Comment (reaganlo) [inline realsense_camera/src/base_nodelet.cpp]: Updated. 
@mdhorn Apparently, I did not remove all the spaces based on the same review comment you had.

**Comment by rjingar** [comment]
Approved :+1:

**Pull request merged**